core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -181,6 +183,14 @@ public void onMatch(RelOptRuleCall call) {
                 .project(project.getProjects(), project.getRowType().getFieldNames())
                 .build();
 
+    // if new filter is always false and none of it's inputs use
+    // correlation variables - simplify new project to empty values
```

Comment:
Why "correlation variables" verification is important?

Reply:
Because without this check, the case from ```sql/sub-query.iq``` : 
```sql
select sal from "scott".emp e
  where 123 IN (
    select cast(null as int)
    from "scott".dept d where e.deptno=d.deptno);
```
Will have final plan : 
```
EnumerableCalc(expr#0..3=[{inputs}], SAL=[$t1])
   EnumerableCorrelate(correlation=[$cor0], joinType=[inner], requiredColumns=[{2}])
     EnumerableCalc(expr#0..7=[{inputs}], EMPNO=[$t0], SAL=[$t5], DEPTNO=[$t7])
       EnumerableTableScan(table=[[scott, EMP]])
     EnumerableValues(tuples=[[]])
```
Though expected one is: 
```
EnumerableValues(tuples=[[]])
```

Reply:
The change seems to be no longer in the PR.

Reply:
It's here, now in ```FilterReduceExpressionsRule.matches(call)``` .